### PR TITLE
[ENH] Changed scitype name `"series-annotator"` to `"detector"`

### LINKED ISSUE #7352: [ENH] rename `"series-annotator"` scitype to `"detector"`
In line with the general rationale of https://github.com/sktime/sktime/issues/7295, we should rename the `"series-annotator"` scitype to `"detector"`.

To ensure downwards compatibility though, it is best to simply *add* the `"detector"` scitype in the list of scitypes (turning str into a list ot str), in the tag `object_type`. This will ensure that queries using the old name will still work, while queries with the new name start working, allowing us to transition to using the latter, new queries in tutorials, primary internal access patterns, etc.

### PR DESCRIPTION
**Pull request by jgyfutub**
#### Reference Issues/PRs
Fixes #7352

Changes scitype name `series-annotator` to `detector`.

Deprecation management:
* `series-annotator` remains present in lookup
* classes now have both types with `series-annotator` as default when coerced to a single type, but the default will change to `detector` in 0.37.0
* 1.0.0 will remove `series-annotator`

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks!

Howeer, we need to ensure that the old tag still works, since users may have that in queries or their code.

For that, I am making some suggestions below.

File: sktime/annotation/base/_base.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ class BaseSeriesAnnotator(BaseEstimator):
     """
 
     _tags = {
-        "object_type": "series-annotator",  # type of object
+        "object_type": "detector",  # type of object
```
Comment (fkiraly) [inline sktime/annotation/base/_base.py:65]: Replacing this by the list `["detector", "series-annotator"]`

File: sktime/tests/test_class_register.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ def get_test_class_registry():
     testclass_dict["global_forecaster"] = TestAllGlobalForecasters
     testclass_dict["param_est"] = TestAllParamFitters
     testclass_dict["regressor"] = TestAllRegressors
-    testclass_dict["series-annotator"] = TestAllAnnotators
```
Comment (fkiraly) [inline sktime/tests/test_class_register.py:63]: leaving both rows

File: sktime/registry/_scitype_coercion.py [hunk 3]
```diff
@@ -16,14 +16,14 @@ def _coerce_transformer_tabular_to_transformer(obj):
...
-def _coerce_series_annotator_to_transformer(obj):
+def _coerce_detector_to_transformer(obj):
     from sktime.annotation.compose._as_transform import AnnotatorAsTransformer
 
     return AnnotatorAsTransformer(obj)
 
 
-_coerce_register[("series-annotator", "transformer")] = (
-    _coerce_series_annotator_to_transformer
+_coerce_register[("detector", "transformer")] = (
```
Comment (fkiraly) [inline sktime/registry/_scitype_coercion.py:25]: leaving both rows

File: sktime/annotation/tests/test_all_annotators.py [hunk 4]
```diff
@@ -29,7 +29,7 @@ class AnnotatorsFixtureGenerator(BaseFixtureGenerator):
     #   additional fixtures, parameters, etc should be added here
     #   TestAllAnnotators should contain the tests only
 
-    estimator_type_filter = "series-annotator"
+    estimator_type_filter = "detectors"
```
Comment (fkiraly) [inline sktime/annotation/tests/test_all_annotators.py:32]: this should be `"detector"`

**Review by fkiraly — COMMENTED** [review]
plus, we should leave `# todo 0.37.0: remove series-annotator` comments next to the changes.

**Comment by jgyfutub** [comment]
@fkiraly thanks for your review! I did all the changes as you requested .Please review the changes.

File: sktime/registry/_base_classes.py [hunk 5]
```diff
@@ -261,11 +261,11 @@ def get_base_class(cls):
         return BaseRegressor
 
 
-class series_annotator(_BaseScitypeOfObject):
+class detector(_BaseScitypeOfObject):  # todo 0.37.0: remove series-annotator
```
Comment (fkiraly) [inline sktime/registry/_base_classes.py:264]: both tags need to be present, although I would list only the `detector` in the API reference. Please also have a look at the tags api reference, in `tags.rst`.

File: sktime/registry/_tags.py [hunk 6]
```diff
@@ -1830,13 +1830,13 @@ class y_inner_mtype(_BaseTag):
     ),
     (
         "task",
-        "series-annotator",
+        "detector",  # todo 0.37.0: remove series-annotator
```
Comment (fkiraly) [inline sktime/registry/_tags.py:1833]: no todo here

File: sktime/annotation/tests/test_all_annotators.py [hunk 7]
```diff
@@ -29,7 +29,7 @@ class AnnotatorsFixtureGenerator(BaseFixtureGenerator):
     #   additional fixtures, parameters, etc should be added here
     #   TestAllAnnotators should contain the tests only
 
-    estimator_type_filter = "series-annotator"
+    estimator_type_filter = "detector"  # todo 0.37.0: remove series-annotator
```
[the remark this answers is not in the document]
Comment (jgyfutub) [reply sktime/annotation/tests/test_all_annotators.py:32]: @fkiraly thanks for review but I have already written `# todo 0.37.0: remove series-annotator`  here  on the right of  `estimator_type_filter = "detector"` ,Can you please refer which todo are you referring to?

File: sktime/registry/_base_classes.py (hunk 5 above)
Comment (jgyfutub) [reply sktime/registry/_base_classes.py:264]: @fkiraly thanks for review, so shall I change ` """Time series annotator."""` to ` """Time series detectorr."""` in line `265`? can you please suggest other tags necessary?

**Comment by jgyfutub** [comment]
@fkiraly can you please check my changes and review it , It has been a long time since I got review after I made changes.Thanks!

**Comment by fkiraly** [comment]
@jgyfutub, thank you for your help! I will now take over this PR to unblock further work on top of it, as previously mentioned.

**Pull request merged**